### DISCUSSION TITLE
Make it possible for skin elements to anchor to other skin elements

### DISCUSSION THREAD

**Discussion by nanashi-1** [ppy/osu#24468] [Ideas] [answered]
It would be great if skin elements can reposition relative to other skin elements instead of the screen. You can make a skin that works on your current display but once that changes, it will be misaligned. It would also be great to export skins that work on any display.

You can observe this when opening the skin editor.

![Screenshot from 2023-08-05 16-34-52](https://github.com/ppy/osu/assets/54083498/032e79dc-86c8-49a5-9ade-8a599561e1d6)

![Screenshot from 2023-08-05 16-34-57](https://github.com/ppy/osu/assets/54083498/bdb8b69b-a7ad-4946-b9b2-222baf04069a)

This is just a suggestion though, there may be other ways to solve this.

**Comment by peppy** [comment]
Could you try using top left anchor for the left pice and top right for the right piece? I think this will already be enough to make this scale correctliy,.

**Reply by nanashi-1** [reply]
I does this instead.

![Screenshot from 2023-08-05 18-05-13](https://github.com/ppy/osu/assets/54083498/46ec041f-5802-4e18-80b9-2c0f627f23c7)
![Screenshot from 2023-08-05 18-05-18](https://github.com/ppy/osu/assets/54083498/7b7c3cd4-c2a6-43e3-88b1-32e4c58d81b7)

**Reply by bdach** [reply, marked as answer]
That makes sense unfortunately and is because we're doing relative sizing but absolute positioning, both of which is what people will want most commonly, but doesn't work for layouts such as the above.
